### PR TITLE
sql/driver: extract method `GoTime()`

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: sql/driver/wire.go [hunk 1]
```diff
@@ -53,9 +54,8 @@ func makeDatum(val driver.Value) (Datum, error) {
 		datum.StringVal = &t
 	case time.Time:
 		// Send absolute time devoid of time-zone.
-		datum.TimeVal = &Datum_Timestamp{
-			Sec:  t.Unix(),
-			Nsec: uint32(t.Nanosecond()),
+		datum.TimestampVal = &proto.Timestamp{
+			WallTime: t.UnixNano(),
```
Comment (petermattis) [inline sql/driver/wire.go]: This isn't the same at all. `time.Time` is a 96-bit value composed of 64-bit seconds and 32-bit nanoseconds field. This compresses it down to a 64-bit value losing tons of range. The 64-bit nanosecond field used by `proto.Timestamp` is fine for our internal purposes (a range of +/- 292 years from 1970), but not for representing a `time.Time`.

Comment (tamird) [inline sql/driver/wire.go]: I'll add some tests to expose this.

**Comment by vivekmenezes** [comment]
while you might be able to use one timestamp for both use cases, I feel that one is internal and the other external. They also seem to have different semantics with the internal one having a virtual time, and being insufficient to express the entire timestamp far into the past/future.

**Comment by tamird** [comment]
PTAL

**Comment by petermattis** [comment]
LGTM

File: sql/driver/wire.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (d Datum) Value() (driver.Value, error) {
 	case *string:
 		val = *t
 	case *Datum_Timestamp:
-		val = time.Unix((*t).Sec, int64((*t).Nsec)).UTC()
+		val = t.GoTime().UTC()
```
Comment (bdarnell) [inline sql/driver/wire.go]: I think the `.UTC()` call ought to be moved into `GoTime()` as well.

**Pull request merged**